cdap-tms/src/main/java/io/cdap/cdap/messaging/client/ClientMessagingService.java

```diff
@@ -288,7 +288,7 @@ private void handleError(int responseCode, Supplier<String> responseBodySupplier
       case HttpURLConnection.HTTP_UNAVAILABLE:
         throw new ServiceUnavailableException(Constants.Service.MESSAGING_SERVICE);
       default:
-        throw new IOException(errorPrefix + ". Reason: " + responseBodySupplier.get());
+        throw new IOException(errorPrefix + ". Http response code: " + responseCode + " .Reason: " + responseBodySupplier.get());
```

Comment:
Instead of encoding it in the message, it's better to introduce a subclass of `IOException` (for backward compatibility) that also implements the `HttpErrorStatusProvider` interface.